beets/importer.py

```diff
@@ -1731,10 +1834,10 @@ def group(item):
         if task.skip:
             continue
         tasks = []
-        sorted_items = sorted(task.items, key=group)
+        sorted_items: list[library.Item] = sorted(task.items, key=group)
         for _, items in itertools.groupby(sorted_items, group):
-            items = list(items)
-            task = ImportTask(task.toppath, [i.path for i in items], items)
+            l_items = list(items)
```

Comment:
This is an interesting one. It seems like originally it was broken: the `items` iterator is emptied by the time it's sent into `ImportTask`. Is that correct?

Reply:
Indeed this could have happen if it truely was an iterator (which it is mostly not), this was an easy catch with the additions to the type hints.

Reply:
As far as I know, `groupby` _always_ returns at iterator as the second returned item. This kind of seems to me that `SingletonimportTask` never ends up using the provided `items`.